pg/wal/recordheader.go

```diff
@@ -46,7 +46,7 @@ func NewRecordHeader(block []byte, location Location, version uint16, reader blo
 
 			if nextPage.IsCont() {
 				cont := nextPage.Continuation()
-				newBlock = append(newBlock, cont[4:]...)
```

Comment:
Yeah I'm literally taking from one to give to the other.  For example:
```
length = 3
start = 20
end = start + length
cont = block[start:end] // 20:23
append cont[4:] // with this we are trying to effectively get block[24:27]
                // but it panics because cont is only 3 bytes

this PR changes the logic to this:

length = 3
start = 24
end = start + length
cont = block[start:end] // 24:27
append cont // now we get block[24:27] without a panic
```